.gitmodules

```diff
@@ -1,91 +1,140 @@
 	path = Sming/Components/rboot/rboot
 	url = https://github.com/raburton/rboot.git
 	ignore = dirty
-[submodule "esptool2"]
-	path = Sming/Components/rboot/esptool2
-	url = https://github.com/raburton/esptool2.git
+[submodule "spiffs"]
+	path = Sming/Components/spiffs/spiffs
+	url = https://github.com/pellepl/spiffs.git
 	ignore = dirty
-[submodule "libsodium"]
-	path = Sming/Libraries/libsodium/libsodium
-	url = https://github.com/jedisct1/libsodium.git
+[submodule "ws_parser"]
+	path = Sming/Components/ws_parser
+	url = https://github.com/charliesome/ws_parser.git
 	ignore = dirty
 
-[submodule "ESP8266.new-pwm"]
-	path = Sming/Arch/Esp8266/Components/driver/new-pwm
-	url = https://github.com/StefanBruens/ESP8266_new_pwm.git
-	ignore = dirty
-[submodule "ESP8266.axtls"]
+
+#
+# `Esp8266` Components
+#
+
+[submodule "Esp8266.axtls"]
 	path = Sming/Components/axtls-8266/axtls-8266
 	url = https://github.com/igrr/axtls-8266.git
 	ignore = dirty
-[submodule "ESP8266.bearssl"]
+[submodule "Esp8266.bearssl"]
 	path = Sming/Components/bearssl-esp8266/bearssl
 	url = https://github.com/earlephilhower/bearssl-esp8266
 	ignore = dirty
-[submodule "ESP8266.umm_malloc"]
-	path = Sming/Arch/Esp8266/Components/heap/umm_malloc
-	url = https://github.com/rhempel/umm_malloc.git
-	ignore = dirty
-[submodule "ESP8266.esp-open-lwip"]
+[submodule "Esp8266.esp-open-lwip"]
 	path = Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip
 	url = https://github.com/pfalcon/esp-open-lwip.git
 	ignore = dirty
-[submodule "ESP8266.lwip2"]
+[submodule "Esp8266.lwip2"]
 	path = Sming/Arch/Esp8266/Components/lwip2/lwip2
 	url = https://github.com/d-a-v/esp82xx-nonos-linklayer.git
 	ignore = dirty
-[submodule "ESP8266.Sdk"]
+[submodule "Esp8266.new-pwm"]
+	path = Sming/Arch/Esp8266/Components/driver/new-pwm
+	url = https://github.com/StefanBruens/ESP8266_new_pwm.git
+	ignore = dirty
+[submodule "Esp8266.umm_malloc"]
+	path = Sming/Arch/Esp8266/Components/heap/umm_malloc
+	url = https://github.com/rhempel/umm_malloc.git
+	ignore = dirty
+[submodule "Esp8266.Sdk"]
 	path = Sming/Arch/Esp8266/Components/esp8266/ESP8266_NONOS_SDK
 	url = https://github.com/espressif/ESP8266_NONOS_SDK.git
 	ignore = dirty
 
+
+#
+# `Esp32` Components
+#
+
+### NONE ###
+
+
+#
+# `Host` Components
+#
+
 [submodule "Host.lwip"]
 	path = Sming/Arch/Host/Components/lwip/lwip
 	url = git://git.savannah.gnu.org/lwip.git
 	ignore = dirty
 
-[submodule "Libraries.Adafruit_ST7735"]
-	path = Sming/Libraries/Adafruit_ST7735
-	url = https://github.com/adafruit/Adafruit-ST7735-Library.git
-	ignore = dirty
-[submodule "Libraries.Adafruit_SSD1306"]
-	path = Sming/Libraries/Adafruit_SSD1306
-	url = https://github.com/adafruit/Adafruit_SSD1306.git
-	ignore = dirty
+
+
+#################################################################################################
+#
+# `Sming/Libraries` - Optional libraries
+#
+# These must be selected using COMPONENT_DEPENDS setting in your component.mk file
```

Comment:
> `Sming/Libraries` - Optional libraries ...
> These must be selected using COMPONENT_DEPENDS

We should start also deprecating the use of ARDUINO_LIBRARIES then. But this can be part of a separate PR.